lib/timeline/component/ItemSet.js

```diff
@@ -1663,16 +1688,18 @@ ItemSet.prototype._onGroupClick = function (event) {
     groupsData = this.groupsData.getDataSet()
   }
 
-  group.showNested = !group.showNested;
+  var nestingGroup = groupsData.get(group.groupId)
+  nestingGroup.showNested = !nestingGroup.showNested;
 
   var nestedGroups = groupsData.get(group.nestedGroups).map(function(nestedGroup) {
     if (nestedGroup.visible == undefined) { nestedGroup.visible = true; }
-    nestedGroup.visible = !!group.showNested;
+    nestedGroup.visible = !!nestingGroup.showNested;
```

Comment:
Why not a direct assignment? `nestingGroup.showNested` is forced to boolean by line `1692`.